ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/RaceConditionIT.scala

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites
+
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.v1.transaction.{TransactionTree, TreeEvent}
+import com.daml.ledger.api.v1.value.RecordField
+import com.daml.ledger.test.semantic.RaceTests._
+import com.daml.lf.data.{Bytes, Ref}
+
+import scala.concurrent.Future
+import scala.util.Success
+
+final class RaceConditionIT extends LedgerTestSuite {
+
+  test(
+    "WWDoubleNonTransientCreate",
+    "Cannot concurrently create multiple non-transient contracts with the same key",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, alice)) =>
+    val Attempts = 100
```

Comment:
I would suggest rephrasing this test. A 100 attempts to archive the same contract is unlikely to expose more problems than doing it 5 times. Creating 5 contracts and archiving them multiple times, otoh might.